pomidor.el

```diff
@@ -486,6 +519,92 @@ TIME may be nil."
     (plist-put state :stopped (current-time)))
   (nconc pomidor-global-state (list (pomidor--make-state))))
 
+(defun pomidor-save-session ()
+  "Save the current session in a file."
+  (interactive)
+  (let ((time-asked-to-save (current-time)))
+    (pomidor-quit)
+    (plist-put (pomidor--current-state) :stopped time-asked-to-save)
+
+    (when (not (file-exists-p pomidor-save-session-file))
+      (with-temp-file pomidor-save-session-file
+        (insert (prin1-to-string (make-hash-table :test 'equal)))))
+    (let* ((file-table (pomidor--read-session))
+           (name (format-time-string "%Y-%m-%dT%H:%M:%S" time-asked-to-save))
+           (global-state (-map (lambda (pomodoro) (plist-put pomodoro :current-time time-asked-to-save))
+                               pomidor-global-state))
+           (global-state (-filter (lambda (pomodoro) (or (plist-get pomodoro :stopped)
+                                                    (plist-get pomodoro :break)
+                                                    (plist-get pomodoro :snooze)))
+                                  pomidor-global-state)))
+      (puthash name global-state file-table)
+      (with-temp-file pomidor-save-session-file
+        (insert (prin1-to-string file-table)))))
+  (message "Pomidor session saved!"))
+
+(defun pomidor-history-previous ()
+  "Move backward in your pomidor history."
+  (interactive)
+  (let* ((session-table (pomidor--read-session))
+         (session-dates (hash-table-keys session-table)))
+    (if (= (length session-dates) 0)
+        (message "You have no session saved.")
+      (let* ((valid-session-dates (if pomidor--current-history-session
```

Comment:
These lines are quite far on the right. Consider introducing `cl-flet` or a helper function?

Reply:
I made a helper function now. You are right! =)